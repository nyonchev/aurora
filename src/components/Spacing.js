import styled from "styled-components";
import PropTypes from "prop-types";
import spacing from "../theme/spacing";
import { mediumAndUp, largeAndUp, smallAndUp } from "../theme/mediaQueries";

const getSpacing = val => spacing[val] || val || 0;

const Spacing = styled.div`
  width: 100%;
  padding-top: ${props => getSpacing(props.top.small)};
  padding-bottom: ${props => getSpacing(props.bottom.small)};

  ${smallAndUp`
    padding-top: ${props => getSpacing(props.top.medium || props.top.small)};
    padding-bottom: ${props =>
      getSpacing(props.bottom.medium || props.bottom.small)};
  `};

  ${mediumAndUp`
    padding-top: ${props =>
      getSpacing(props.top.large || props.top.medium || props.top.small)};
    padding-bottom: ${props =>
      getSpacing(
        props.bottom.large || props.bottom.medium || props.bottom.small
      )};
  `};

  ${largeAndUp`
    padding-top: ${props =>
      getSpacing(
        props.top.xLarge ||
          props.top.large ||
          props.top.medium ||
          props.top.small
      )};
    padding-bottom: ${props =>
      getSpacing(
        props.bottom.xLarge ||
          props.bottom.large ||
          props.bottom.medium ||
          props.bottom.small
      )}
  `};
`;

const spacingShape = PropTypes.shape({
  small: PropTypes.string,
  medium: PropTypes.string,
  large: PropTypes.string,
  xLarge: PropTypes.string
});

Spacing.propTypes = {
  top: spacingShape,
  bottom: spacingShape
};

Spacing.defaultProps = {
  top: {},
  bottom: {}
};

export default Spacing;
