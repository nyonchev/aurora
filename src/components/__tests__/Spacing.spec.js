import React from "react";
import renderer from "react-test-renderer";
import Spacing from "../Spacing";

describe("Spacing", () => {
  it("renders top and bottom spacing for xSmall device (320-479)", () =>
    expect(
      renderComponent({
        top: { xSmall: "normal" },
        bottom: { xSmall: "normal" }
      })
    ).toMatchSnapshot());

  it("renders top and bottom spacing for small device (480-767)", () =>
    expect(
      renderComponent({ top: { small: "normal" }, bottom: { small: "normal" } })
    ).toMatchSnapshot());

  it("renders top and bottom spacing for medium device (767-1023)", () =>
    expect(
      renderComponent({
        top: { medium: "normal" },
        bottom: { medium: "normal" }
      })
    ).toMatchSnapshot());

  it("renders top and bottom spacing for large device (>1024)", () =>
    expect(
      renderComponent({
        top: { large: "normal" },
        bottom: { large: "normal" }
      })
    ).toMatchSnapshot());

  function renderComponent(props = {}) {
    return renderer.create(<Spacing {...props} />).toJSON();
  }
});
