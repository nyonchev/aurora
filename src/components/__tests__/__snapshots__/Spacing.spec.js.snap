// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spacing renders top and bottom spacing for large device (>1024) 1`] = `
.c0 {
  width: 100%;
  padding-top: 0;
  padding-bottom: 0;
}

@media screen and (min-width:480px) {
  .c0 {
    padding-top: 0;
    padding-bottom: 0;
  }
}

@media screen and (min-width:768px) {
  .c0 {
    padding-top: 0;
    padding-bottom: 0;
  }
}

@media screen and (min-width:1024px) {
  .c0 {
    padding-top: 24px;
    padding-bottom: 24px;
  }
}

<div
  className="c0"
/>
`;

exports[`Spacing renders top and bottom spacing for medium device (767-1023) 1`] = `
.c0 {
  width: 100%;
  padding-top: 0;
  padding-bottom: 0;
}

@media screen and (min-width:480px) {
  .c0 {
    padding-top: 0;
    padding-bottom: 0;
  }
}

@media screen and (min-width:768px) {
  .c0 {
    padding-top: 24px;
    padding-bottom: 24px;
  }
}

@media screen and (min-width:1024px) {
  .c0 {
    padding-top: 24px;
    padding-bottom: 24px;
  }
}

<div
  className="c0"
/>
`;

exports[`Spacing renders top and bottom spacing for small device (480-767) 1`] = `
.c0 {
  width: 100%;
  padding-top: 0;
  padding-bottom: 0;
}

@media screen and (min-width:480px) {
  .c0 {
    padding-top: 24px;
    padding-bottom: 24px;
  }
}

@media screen and (min-width:768px) {
  .c0 {
    padding-top: 24px;
    padding-bottom: 24px;
  }
}

@media screen and (min-width:1024px) {
  .c0 {
    padding-top: 24px;
    padding-bottom: 24px;
  }
}

<div
  className="c0"
/>
`;

exports[`Spacing renders top and bottom spacing for xSmall device (320-479) 1`] = `
.c0 {
  width: 100%;
  padding-top: 24px;
  padding-bottom: 24px;
}

@media screen and (min-width:480px) {
  .c0 {
    padding-top: 24px;
    padding-bottom: 24px;
  }
}

@media screen and (min-width:768px) {
  .c0 {
    padding-top: 24px;
    padding-bottom: 24px;
  }
}

@media screen and (min-width:1024px) {
  .c0 {
    padding-top: 24px;
    padding-bottom: 24px;
  }
}

<div
  className="c0"
/>
`;
